src/ethereum_test_specs/blockchain.py

```diff
@@ -692,7 +710,10 @@ def make_hive_fixture(
         """Create a hive fixture from the blocktest definition."""
         fixture_payloads: List[FixtureEngineNewPayload] = []
 
-        pre, genesis = BlockchainTest.make_genesis(self.genesis_environment, self.pre, fork)
+        pre, genesis = self.make_genesis(
+            fork=fork,
+            apply_pre_allocation_blockchain=fixture_format != BlockchainEngineXFixture,
```

Comment:
The reason behind this is that, `BlockchainEngineFixture` receives the environment directly from the `genesis_environment` field as it comes from the test, and the tests do not set the fork pre-required system contracts.
On the other hand, `BlockchainEngineXFixture` receives the environment from the pre-allocation group, where the `fork.pre_allocation_blockchain` has already been applied before saving to file (and loading again during this step).